charts/jenkins/templates/cache-cronjobs.yaml

```diff
@@ -1,38 +1,32 @@
-{{- /* create map of agents by clearJob.componentName */}}
-{{- if .Values.agent.cache.clearJob.create }}
-  {{- $_ := set .Values "agentMap" (dict .Values.agent.cache.clearJob.componentName .Values.agent) }}
-{{- else }}
-  {{- $_ := set .Values "agentMap" (dict) }}
-{{- end }}
+{{- /* create agent map */}}

```

Comment:
I'm recommending a map based on agent names as I'm envisioning moving the merge logic to a `createAgentMap` template and using it throughout the code. The template will generate the agent map and make it available as `.Values.agents`.